docs/internals/controllers.md

```diff
@@ -0,0 +1,34 @@
+# Controllers
+
+The main controller implementations ([cluster] and [machine]) are located
+within  the `cluster-api` library, and each perform almost identical basic
+steps within their `Reconcile` methods. The controllers have the responsibility
+of watching for changes to specific resource types and reacting to those
+changes in order to reach a desired state by calling out to the necessary
+actuator methods.
+
+  - Controller watches for changes on a resource type
+  - A change on a watched resource type is observed and the controller
+    receives a reconcile request
+  - An attempt is made to fetch the appropriate object from Kubernetes; and if
```

Comment:
Is this still true after switching to CRDs? I thought the framework took care of this part now.

Reply:
This does still appear to be the case in the [machine/controller.go](https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machine/controller.go) and [cluster/controller.go](https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/cluster/controller.go) files.